### ISSUE TITLE
Command-line export

### ISSUE THREAD

**Issue by Timmmm** [shd101wyy/markdown-preview-enhanced#181]
Is there a ways to convert a markdown file to HTML from the command line? Or even better, a directory of markdown files?

This would allow us to store documentation in Git, and automatically render it to HTML so people without Atom & this plugin can still see the result.

(The plugin is amazing btw!)

**Comment by shd101wyy** [comment]
Hi @Timmmm , 

Right now command-line export is not supported.  
I don't have a plan to implement command-line support, but I will let you know if I decide to support.

Thank you

**Comment by Timmmm** [comment]
Ok. If I wanted to implement this, can you give me a rough guide of how to do it? Is the rendering part of the library well separated from Atom? Can it be extracted into a standalone program? (That would be extra awesome because then I can make a Stash plugin for it.)

**Comment by shd101wyy** [comment]
Yes it is possible, but will be hard because some parts are related to [MarkdownPreviewEnhancedView](https://github.com/shd101wyy/markdown-preview-enhanced/blob/master/lib/markdown-preview-enhanced-view.coffee#L20) class.   
You may need to read through the following files:  
* [md.coffee](https://github.com/shd101wyy/markdown-preview-enhanced/blob/master/lib/md.coffee)    
This file include how we parse markdown content, math typesettings, resolve image paths, generate code blocks & code chunks (I think you can ignore the code chunk for now).  
* [markdown-preview-enhanced-view.coffee](https://github.com/shd101wyy/markdown-preview-enhanced/blob/master/lib/markdown-preview-enhanced-view.coffee)  and the following functions
    * saveAsPDF & printPDF & getHTMLContent
    * phantomJSExport
    * pandocDocumentExport
    * generateEbook  
* [ebook-convert.coffee](https://github.com/shd101wyy/markdown-preview-enhanced/blob/master/lib/ebook-convert.coffee)  
This file provide API to generate ebook files.  You can use it directly
* [pandoc-wrapper.coffee](https://github.com/shd101wyy/markdown-preview-enhanced/blob/master/lib/pandoc-wrapper.coffee)  
This file provide API to use pandoc. However you may need to modify the functions.    


My code is actually messy, so ask me if you meet any problem.  
Thank you.

**Comment by xu4wang** [comment]
+1 
@Timmmm please let me know when you start to work on this. 
I was thinking to make a web based wiki, use the same file format with markdown-preview-enhanced.

**Comment by Timmmm** [comment]
We decided not to use it in the end. We were thinking of having
documentation in a git repo, and automatically rendering it. But some other
people really wanted to use Confluence so we are going with that and using
its PlantUML plugin. Not as nice IMO but I am not the boss unfortunately.

**Comment by aerophagiano** [comment]
+1
I think that a cli driven export would be a killer feature!
By now I have to rely on pandoc for an automated "build" of all my md docs :) but I'm missing many cool features (i.e. the awesome "@import")  Do you have any hint on how to use pandoc (or similar) to batch translate markdown to html with that feature?

**Comment by shd101wyy** [comment]
I would recommend you to check [mume project](https://github.com/shd101wyy/mume), which powers the markdown preview enhanced package. 
You can use that library to build command line tool.

**Comment by aerophagiano** [comment]
oh gosh! Exactly what I was looking for! Awesome.

**Comment by penseleit** [comment]
@aerophagiano, did you implement a CLI using Mume? I would be interested in having a look at it for my own use, if possible.

**Comment by penseleit** [comment]
OK, I got mume to work for automatic processing of my Markdown. 

I wrote a Javascript file to process all of the .md files in the current folder. I run this script via Visual Studio Code's tasks functionality

Code below (improve it if you want to):

``` javascript

// node.js
const mume = require('C:/src/mume-lib/node_modules/@shd101wyy/mume'); //This is the path where I downloaded the mume libraries - modify this to suit your own path

// es6
// import * as mume from "@shd101wyy/mume"

async function main() {
  await mume.init();

  const engine = new mume.MarkdownEngine({
    filePath: "Help.md",
    config: {
      previewTheme: "github-light.css",
      codeBlockTheme: "default.css",
      printBackground: false,
      enableScriptExecution: true, // <= for running code chunks
    },
  });

var fs = require('fs');
 
var path = '.'; //current folder
var mdextension = '.md';

try { 

  var mdfiles = fs.readdirSync(path);
  if (typeof mdfiles !== 'undefined' && mdfiles.length > 0) {
  
    for(var i=0; i < mdfiles.length; i++)
    {
      var mdindex = mdfiles[i].indexOf(mdextension, (mdfiles[i].length - 3));
      if (mdindex  !== -1)
      {
        console.log('Processing: ' + mdfiles[i]);
        engine.filePath = mdfiles[i];
        
        // html export
        await engine.htmlExport({ offline: true, runAllCodeChunks: true });
      }
    }
  }
}
catch(e)
{
  console.log(e.name + ': ' + e.message);
}
 return process.exit();
}
main();
``

**Comment by pengyuwei** [comment]
The ability to `export HTML` from the `command line` is very much needed.
I currently use VSCode to record documentation notes in Markdown format, markdown-preview-enhanced preview MD, and `Pandoc` to generate HTML, which is automatically deployed to the Web server via scripts, and the entire process is automated.
However, I would like to use the Markdown-preview -enhanced to generate the HTML, but that would require manually generating it one file at a time. If markdown-preview-enhanced **provides a command line**, you can cascade the entire automated process.

**Comment by shd101wyy** [comment]
Please check https://github.com/shd101wyy/mume
